R/get_entity.R

```diff
@@ -132,9 +132,53 @@
-  inferred_entity_ids <- infer_entity_type(entity_id=entity_id, entity_list=NULL) # Infers the entity schemas for each element
-  entity_single_get_endpoints  <- .get_api_endpoints(entity_id=inferred_entity_ids, entity_list=NULL) # Extracts the Single-Get API endpoint URL format for each entity identifier according to its respective schema element
-  api_queries <- .craft_api_queries(entity_single_get_endpoints=entity_single_get_endpoints) # Rewrites the Single-Get API endpoint URL formats into proper URL strings for each identifier
-  response_list <- .submit_queries(api_queries=api_queries, api_key=api_key) # Uses the URL strings to call the entity identifier's Single-Get API endpoint in Benchling and extracts the contents of the response
+  inferred_entity_ids <- infer_entity_type(entity_id=entity_id, entity_list=NULL) 
+    # Infers the entity schemas for each element.
+  invalid_entity_ids1 <- names(inferred_entity_ids[which(is.na(inferred_entity_ids))]) 
+    # Finds all the invalid entity identifiers whose values are NA.
+  if (length(invalid_entity_ids1) > 0 & 
+      length(invalid_entity_ids1) < length(inferred_entity_ids)) { 
+    # Checks for any invalid entity identifiers while making sure that every
+    # entity identifier is not invalid.
+    warning(glue::glue("The following entity identifier(s) are invalid and will be removed: {toString(invalid_entity_ids1)}."))
+    # Generates a warning for any invalid entity identifiers found.
+    inferred_entity_ids <- inferred_entity_ids[! names(inferred_entity_ids) 
+                                               %in% invalid_entity_ids1]
+    # Removes the invalid entity identifiers from inferred_entity_ids.
+  } else if (length(invalid_entity_ids1) == length(inferred_entity_ids)) {
+    # Checks if all the entity identifiers are invalid.
+    stop("All entity identifiers are invalid.")
+    # Stops the function.
+  }
+  entity_single_get_endpoints  <- .get_api_endpoints(entity_id=inferred_entity_ids, 
+                                                     entity_list=NULL) 
+    # Extracts the Single-Get API endpoint URL format for each entity identifier 
+    # according to its respective schema element.
+  api_queries <- .craft_api_queries(entity_single_get_endpoints=
+                                    entity_single_get_endpoints) 
+    # Rewrites the Single-Get API endpoint URL formats into proper URL 
+    # strings for each identifier.
+  response_list <- .submit_queries(api_queries=api_queries, 
+                                   api_key=api_key) 
+    # Uses the URL strings to call the entity identifier's Single-Get 
+    # API endpoint in Benchling and extracts the contents of the response.
+  invalid_entity_ids2 <- names(response_list[which(purrr::map(response_list, 
+                                                   ~ !is.null(.$error)) == TRUE)])
+    # Finds all the entity identifiers that generated an error in their response.
+  if (length(invalid_entity_ids2) > 0 & 
```

Comment:
Oh okay. But even though I state why the request failed based on the API response, it should still be removed from the output right?

I think I do not to make another function to accomplish this as I think minimal work is required to place this check inside the get_entity function.

Reply:
@AK197624 You want to print the message from the error response, not hard-code in the reason why the request didn't work.

Reply:
Oh okay then.

Reply:
Should the warning message for each entity identifier with a request failure be formatted saying "Error in response output for {entity identifier}. {error message}" or something similar?

Reply:
But what if all of the entity identifiers have an error in their response. In addition to the function stopping, should I also have the function generate individual error responses so that the user what the error is exactly for each entity identifier?